genai/client.go

```diff
@@ -146,7 +146,7 @@ func (m *GenerativeModel) generateContent(ctx context.Context, req *pb.GenerateC
 	}
 	for {
 		_, err := iter.Next()
-		if err == iterator.Done {
+		if err == iterator.Done && iter.merged != nil {
```

Comment:
I wonder if this is the right place to fix, or would that be in `Next` where `nil` is returned - which can happen without error only at the last line?

Reply:
@jba for thoughts on this

Reply:
I think we want to catch this even deeper. See #68.